### PR TITLE
register actinia modules on startup

### PR DESCRIPTION
**Pull request by mmacata**
To exploit more functionality of actinia and therefore of GRASS GIS, all modules (grass-modules and actinia-modules) are registered on startup with this PR. This way they appear as processes under `/processes` endpoint. For now, only "id", "description" and "categories" are shown for external processes (depending on actinia-gdi endpoint).

Snippet of response containing default openeo process "variance", grass-module "r.quantile" and actinia-module "slope_aspect":

```
    {
      "deprecated": false,
      "description": "Computes the sample variance of an array of numbers by calculating the square of the standard deviation (see ``sd()``).",
      "experimental": false,
      "id": "variance",
      "links": [],
      "parameters": [
        {
          "deprecated": false,
          "description": "An array of numbers.",
          "experimental": false,
          "name": "data",
          "optional": false,
          "schema": {
            "items": {
              "type": [
                "number",
                "null"
              ]
            },
            "type": "array"
          }
        },
        {
          "default": true,
          "deprecated": false,
          "description": "Indicates whether no-data values are ignored or not.",
          "experimental": false,
          "name": "ignore_nodata",
          "optional": true,
          "schema": {
            "type": "boolean"
          }
        }
      ],
      "returns": {
        "description": "The computed sample variance.",
        "media_type": "application/json",
        "schema": {
          "type": [
            "number",
            "null"
          ]
        }
      },
      "summary": "Variance"
    },
    {
      "categories": [
        "algebra",
        "grass-module",
        "percentile",
        "quantile",
        "raster",
        "statistics"
      ],
      "description": "Compute quantiles using two passes.",
      "id": "r.quantile"
    },
    {
      "categories": [
        "actinia-module"
      ],
      "description": "Calculates slope and aspect of elevation map.",
      "id": "slope_aspect"
    },
```

### CODE REVIEW

File: src/openeo_grass_gis_driver/register_processes.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+def register_processes():
+
+    iface = ActiniaInterface()
+    iface.set_auth(ActiniaConfig.USER, ActiniaConfig.PASSWORD)
+    status_code, modules = iface.list_modules()
+
+    if status_code == 200:
+        for module in modules:
+            # TODO: add logger
+            print("registering %s" % module['id'])
```
Comment (metzm) [inline src/openeo_grass_gis_driver/register_processes.py:25]:
```suggestion
            # print("registering %s" % module['id'])
```

File: src/openeo_grass_gis_driver/register_processes.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            PROCESS_DESCRIPTION_DICT[module['id']] = module
```
Comment (metzm) [inline src/openeo_grass_gis_driver/register_processes.py:26]: Maybe use a separate dict like a new ACTINIA_PROCESS_DESCRIPTION_DICT to easily distinguish between native openeo processes and additional actinia processes?

**Review by anikaweinmann — APPROVED** [review]

File: src/openeo_grass_gis_driver/register_processes.py (hunk 1 above)
Comment (mmacata) [reply src/openeo_grass_gis_driver/register_processes.py]: done in bb0cc08

File: src/openeo_grass_gis_driver/register_processes.py (hunk 2 above)
Comment (mmacata) [reply src/openeo_grass_gis_driver/register_processes.py]: Good idea. Done in bb0cc08

**Review by metzm — APPROVED** [review]
Looks good and works for me.

**Review by neteler — APPROVED** [review]

**Comment by mmacata** [comment]
Then I will merge now

**Pull request merged**